src/ai-dialog-header.ts

```diff
@@ -8,7 +8,7 @@ import {DialogController} from './dialog-controller';
       type="button"
       class="dialog-close"
       aria-label="Close"
-      if.bind="!controller.settings.lock"
+      show.bind="showCloseButton"
```

Comment:
Generally I prefer using `if` over `show` unless there are descendants as it keep the DOM a bit cleaner but works the same either way just throwing it out there.